doxia-core/src/main/java/org/apache/maven/doxia/index/IndexingSink.java

```diff
@@ -150,14 +158,20 @@ public void text(String text, SinkEventAttributes attributes) {
                 case SECTION_3:
                 case SECTION_4:
                 case SECTION_5:
+                case SECTION_6:
                     // -----------------------------------------------------------------------
                     // Sanitize the id. The most important step is to remove any blanks
                     // -----------------------------------------------------------------------
 
                     // append text to current entry
                     IndexEntry entry = stack.lastElement();
 
-                    String title = entry.getTitle() + text;
+                    String title;
```

Comment:
Why not `title = entry.getTitle()` and then `title += text`/`title = text`?

Reply:
Because entry.getTitle() might return null

Reply:
I mean for the if else clause:
```
String title = entry.getTitle();
                    if (title != null) {
                        title += text;
                    } else {
                        title = text;
                    }
```

Reply:
Done in https://github.com/apache/maven-doxia/pull/228/commits/bd1e61ea7705af4a6a6f19ad28355a5c6807c343.